### PR TITLE
Refactor test conditions, stop running SQL tests on 4.0 server

### PR DESCRIPTION
**Pull request by zpqrtbnk**
We realized that our SQL tests run against 4.0 servers, which makes no sense, as SQL is not supported there (only beta) and jet is n/a. Thus, it breaks our QA test suites. We *do* have a condition mechanism to filter tests per server version, but up until now it was relying on an environment variable set by `hz.ps1`, not the actual version of the running server.

With this PR,
* We fetch the server version from the server. Therefore, we are sure it is correct
* We remove the environment variable support as it does not make sense anymore
* If the RC is already running, we validate that the version matches the expected version
* We enforce the `[5.0,)` condition on all SQL tests = run them on 5.0 and above

### CODE REVIEW

**Review by emreyigit — APPROVED** [review]

**Comment by alisengul53** [comment]
>We realized that our SQL tests run against 4.0 servers, which makes no sense, as SQL is not supported there (only beta)

Minor: SQL comes with 4.1

**Comment by alisengul53** [comment]
The failures for Build PR, are they expected?

**Comment by alisengul53** [comment]
When I try it locally, it seems it is still sending 5.x server config to rc and trying to create 4.x server with this config. So tests are fails immediately instead of skipping because one time set up is failed.
Or did I miss something?

**Comment by zpqrtbnk** [comment]
Failures are *not* expected, investigating.

File: src/Hazelcast.Net.Tests/Testing/NUnitPropertyAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            Assert.IsNull(TestContext.CurrentContext.Test.Properties["FIXTURE"]?.FirstOrDefault());
+
+            // test method properties are immediately available
+            Assert.AreEqual("method", TestContext.CurrentContext.Test.Properties["METHOD"]?.FirstOrDefault());
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        [Property("METHOD", "method")]
+        public void CanSetAndGetTestSuiteProperties(int ignored)
```
Comment (alisengul53) [inline src/Hazelcast.Net.Tests/Testing/NUnitPropertyAttributeTests.cs:54]: Do we have a purpose of this parameter?

File: src/Hazelcast.Net.Tests/Testing/NUnitTestActionAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            }
+
+            Assert.That(TestContext.CurrentContext.Test.Properties["FIXTURE"]?.FirstOrDefault(), Is.EqualTo("fixture"));
+            Assert.That(TestContext.CurrentContext.Test.Properties["TEST"]?.FirstOrDefault(), Is.EqualTo("test"));
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        [Some("TEST", "test")]
+        public void TestSuite(int ignored)
```
Comment (alisengul53) [inline src/Hazelcast.Net.Tests/Testing/NUnitTestActionAttributeTests.cs:49]: Same question here

Comment (zpqrtbnk) [reply src/Hazelcast.Net.Tests/Testing/NUnitTestActionAttributeTests.cs:49]: Same answer here

**Comment by zpqrtbnk** [comment]
> The failures for Build PR, are they expected?

Not happening anymore, were probably temporary glitches.

**Comment by zpqrtbnk** [comment]
> When I try it locally, it seems it is still sending 5.x server config to rc and trying to create 4.x server with this config. So tests are fails immediately instead of skipping because one time set up is failed.

Interesting. With this PR we should *not* run v5 tests against v4 servers. However... At the moment we still use one unique server configuration file for both. I need to discuss it with you to understand the situation better.

**Comment by alisengul53** [comment]
> > When I try it locally, it seems it is still sending 5.x server config to rc and trying to create 4.x server with this config. So tests are fails immediately instead of skipping because one time set up is failed.
> 
> Interesting. With this PR we should _not_ run v5 tests against v4 servers. However... At the moment we still use one unique server configuration file for both. I need to discuss it with you to understand the situation better.

Most probably It was my mistake, a later tried worked expected.

**Comment by zpqrtbnk** [comment]
OK, there *were* issues when running against v4 server due to `FencedLock` tests, I have fixed these tests and now one can run the whole tests against v4 or v5 and it works (e.g. `hz test -server-version 4.0`). Plus, we have `FencedLock` tests against 3 members, which is better.

**Comment by alisengul53** [comment]
Works well now.

**Review by alisengul53 — APPROVED** [review]

**Review by emreyigit — APPROVED** [review]
👌

**Pull request merged**